src/main/java/io/cdap/e2e/pages/actions/CdfPipelineRunAction.java

```
@@ -52,4 +56,22 @@ public static Boolean isProvisioning(){
         }
         return bool;
     }
+
+    public static String captureRawLogs() {
+        try {
+
+            JavascriptExecutor js = (JavascriptExecutor) SeleniumDriver.getDriver();
+            js.executeScript("arguments[0].click()", cdfPipelineRunLocators.logsArrow);
+            cdfPipelineRunLocators.viewRawLogs.click();
+            String parent = SeleniumDriver.getDriver().getWindowHandle();
+            ArrayList<String> tabs2 = new ArrayList<String>(SeleniumDriver.getDriver().getWindowHandles());
+            SeleniumDriver.getDriver().switchTo().window(tabs2.get(1));
+            String logs = SeleniumDriver.getDriver().findElement(By.xpath("/html/body/pre")).getText();
+            SeleniumDriver.getDriver().close();
+            SeleniumDriver.getDriver().switchTo().window(parent);
+            return logs;
+        } catch (Exception e) {
+          return "*****Error in Capturing logs:+"+e;
```

It's important. If we fail to capture logs, let it fail the test. Otherwise we can end up in silently not having logs and see it only when we need the logs